content/blog/the-rise-of-special-project-infra.md

```diff
@@ -0,0 +1,168 @@
+    - User name: `admin`
+    - Password: `<stream key>` (`abc123` by default)
+
+1.  In server settings, set a new Stream Key. (Note that this is also the administrative web UI password.)
+
+1.  In general settings, modify instance details, tags, and social handles as appropriate.
+
+1.  In video settings:
+
+    - Add stream variants to enable adaptive bitrate streaming to accommodate users on various-quality networks. The following table may be used as a guideline:
+
+      | Encoder | Name | Resolution (WxH) |        Bitrate | Framerate    |
+      | ------- | ---- | ---------------- | -------------: | ------------ |
+      | x264    | SD   | 854x480          |  800-1200 kbps | 25 or 30 fps |
+      | x264    | HD   | 1280x720         | 1200-1900 kbps | 25 or 30 fps |
+      | x264    | FHD  | 1920x1080        | 1900-4500 kbps | 25 or 30 fps |
+
+      Note: some services recommend a higher bitrate for HD:
+
+      | Encoder | Name | Resolution (WxH) |   Bitrate | Framerate    |
+      | ------- | ---- | ---------------- | --------: | ------------ |
+      | x264    | HD   | 1280x720         | 3000 kbps | 25 or 30 fps |
+      | x264    | FHD  | 1920x1080        | 4500 kbps | 25 or 30 fps |
+
+      Note: our stress tests showed that the use of standard variants with our VPS were reportedly adequate for most users but problematic for some in Japan, Thailand, and the UK. Based on these results, object storage and CDN caching are recommended.
+
+      Additional notes:
+
+      - For interactive live streams (e.g., lectures with Q&A sessions), consider decreasing the latency buffer to Low.
+      - For further tuning, consider referring to a bitrate calculator (e.g., [Bitrate Calc](https://bitratecalc.com/)).
+
+#### Streaming client configuration
+
+With the server side all set, it was time to get streaming! The basic procedure follows:
+
+1. On a local workstation, install and launch [OBS Studio](https://obsproject.com/) ([obs-studio](https://search.nixos.org/packages?channel=22.05&show=obs-studio&from=0&size=50&sort=relevance&type=options&query=obs-studio) in Nixpkgs).
+
+1. In stream settings, set the following values:
+
+   - Service: `Custom...`
+   - Server: `rtmp://<target FQDN>` (e.g., `rtmp://live.nixos.org`)
+   - Stream key: `abc123` (by default)
+
+1. Go live.
+
+1. Confirm the stream at the designated URL (`https://live.nixos.org` in our case).
+
+_It worked?_ It worked. Crisis averted with time to spare. 🍹
+
+## Next Steps
+
+Following a successful first lecture, the SoN infra team is concerned not only with maintenance and optimization of the infrastructure we’ve already deployed but also with innovation on a larger scale. The ultimate goal is to contribute a fully self-hosted, documented, NixOS-based, general-purpose, FLOSS[^floss] conference suite that not only is usable for the SoN Public Lecture Series but also can scale for a modern conference experience—namely [NixCon](https://nixcon.org/).
```

Comment:
```suggestion
Following a successful first lecture, the SoN infra team is concerned not only with maintenance and optimization of the infrastructure we’ve already deployed but also with innovation on a larger scale. The ultimate goal is to contribute a fully self-hosted, documented, NixOS-based, general-purpose, FLOSS[^floss] conference suite that not only is usable for the SoN Public Lecture Series, but also can scale for a modern conference experience—namely [NixCon](https://nixcon.org/).
```

Reply:
In this sentence, “that” serves as the subject of a declarative content clause. My understanding is that introducing a comma here without a subject would violate the parallel structure. While this could be corrected by simply repeating the subject – `...suite that not only is A, but that also can do B.` – the redundancy feels a bit awkward.

In any case, I’m happy to rework the sentence if it feels too long as-is.

Reply:
Yes please, it was hard to figure out. I'm not all that proficient with English grammar, I only care about it being easy to read.

Reply:
👍